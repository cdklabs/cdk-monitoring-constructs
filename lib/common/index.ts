export * from "./alarm";
export * from "./metric";
export * from "./monitoring";
export * from "./url";
export * from "./widget";
export * from "./strings";
