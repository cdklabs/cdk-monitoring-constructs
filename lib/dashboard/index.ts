export * from "./BitmapDashboard";
export * from "./DashboardRenderingPreference";
export * from "./DynamicDashboardSegment";
export * from "./DashboardSegment";
export * from "./DashboardWithBitmapCopy";
export * from "./DefaultDashboardFactory";
export * from "./DynamicDashboardFactory";
export * from "./DefaultWidgetFactory";
export * from "./IDashboardFactory";
export * from "./IDynamicDashboardFactory";
export * from "./IWidgetFactory";
export * from "./MonitoringNamingStrategy";
export * from "./SingleWidgetDashboardSegment";
export * from "./widget";
