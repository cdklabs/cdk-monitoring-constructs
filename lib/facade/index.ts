export * from "./IMonitoringAspect";
export * from "./MonitoringFacade";
export * from "./MonitoringAspect";
