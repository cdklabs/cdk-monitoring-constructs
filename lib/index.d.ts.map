{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,UAAU,CAAC;AACzB,cAAc,aAAa,CAAC;AAC5B,cAAc,UAAU,CAAC;AACzB,cAAc,cAAc,CAAC"}