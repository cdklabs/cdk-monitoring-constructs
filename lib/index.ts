export * from "./common";
export * from "./dashboard";
export * from "./facade";
export * from "./monitoring";
