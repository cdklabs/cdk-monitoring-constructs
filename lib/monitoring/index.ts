export * from "./aws-acm";
export * from "./aws-apigateway";
export * from "./aws-apigatewayv2";
export * from "./aws-appsync";
export * from "./aws-billing";
export * from "./aws-cloudfront";
export * from "./aws-cloudwatch";
export * from "./aws-codebuild";
export * from "./aws-docdb";
export * from "./aws-dynamo";
export * from "./aws-ec2";
export * from "./aws-ecs-patterns";
export * from "./aws-elasticache";
export * from "./aws-glue";
export * from "./aws-kinesis";
export * from "./aws-kinesisanalytics";
export * from "./aws-lambda";
export * from "./aws-loadbalancing";
export * from "./aws-opensearch";
export * from "./aws-rds";
export * from "./aws-redshift";
export * from "./aws-s3";
export * from "./aws-secretsmanager";
export * from "./aws-sns";
export * from "./aws-sqs";
export * from "./aws-step-functions";
export * from "./aws-synthetics";
export * from "./aws-wafv2";
export * from "./custom";
export * from "./fluentbit";
