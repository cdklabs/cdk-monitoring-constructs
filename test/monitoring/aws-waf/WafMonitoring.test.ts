import { Stack } from "monocdk";
import { Template } from "monocdk/assertions";

import { WafMonitoring } from "../../../lib";
import { TestMonitoringScope } from "../TestMonitoringScope";

test("snapshot test: no alarms", () => {
  const stack = new Stack();

  const scope = new TestMonitoringScope(stack, "Scope");

  new WafMonitoring(scope, { aclName: "DummyAcl" });

  expect(Template.fromStack(stack)).toMatchSnapshot();
});
