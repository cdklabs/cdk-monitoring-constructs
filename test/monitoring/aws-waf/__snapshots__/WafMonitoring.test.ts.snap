// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: no alarms 1`] = `Object {}`;
